pkg/apis/pipeline/v1alpha1/workspace_validation.go

```diff
@@ -0,0 +1,48 @@
+/*
+Copyright 2019 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"context"
+
+	"knative.dev/pkg/apis"
+)
+
+// Validate looks at the Volume provided in wb and makes sure that it is valid.
+// This means that the only one VolumeSource can be specified, and also that the
+// supported VolumeSource is itself valid.
+func (b *WorkspaceBinding) Validate(ctx context.Context) *apis.FieldError {
+	if b == nil {
```

Comment:
hrm. i'm not sure how a WorkspaceBinding could have its Validate method called but also be nil - I would've thought this is an error since I don't think there's a usecase where it should happen during normal operation.  iow what would the YAML look like where the binding is nil but we consider it a valid entry in the workspace binding list?  If this is a submission from an API client I'd almost certainly expect to get back an error if I submitted a WorkspaceBinding list with nil entries.  Not sure, wdyt?

Reply:
🤯

Reply:
you make a very good point! i almost want to drop the check all together but also i feel like not handling that case usually leads to a panic at some point

it seems like most validate functions include something like this:

```go
	if equality.Semantic.DeepEqual(ts, &TaskSpec{}) {
		return apis.ErrMissingField(apis.CurrentField)
	}
```

so maybe that'll do it - im guessing a nil pointer is semantically deep equal to an empty binding - and if you somehow managed to provide that, it's probably an error?

Reply:
update: its not

Reply:
so now 1) doing the semantic check and 2) treating a failure of that _or_ a nil value as an error, what do you think?

Reply:
yeah that sounds like a good approach to me!